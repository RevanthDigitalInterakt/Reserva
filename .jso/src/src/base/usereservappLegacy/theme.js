  Object.defineProperty(exports, "__esModule", {
    value: true
  });
  exports.theme = undefined;
  var tokens = _interopRequireWildcard(_$$_REQUIRE(_dependencyMap[0]));
  function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
  function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
  var theme = exports.theme = {
    space: {
      quarck: 4,
      nano: 8,
      micro: 12,
      xxxs: 16,
      xxs: 24,
      xs: 32,
      sm: 40,
      md: 48,
      xl: 64,
      xxl: 80
    },
    shadows: {
      bottomBarShadow: '0px -3px 2px rgba(0,0,0,0.1)',
      topBarShadow: '0px 3px 2px rgba(0,0,0,0.1)'
    },
    colors: {
      preto: tokens.preto,
      vermelhoFechadoRSV: tokens.vermelhoFechadoRSV,
      marinho: tokens.marinho,
      neutroFrio2: tokens.neutroFrio2,
      neutroQuente2: tokens.neutroQuente2,
      vermelhoRSV: tokens.vermelhoRSV,
      verde: tokens.verde,
      neutroQuente1: tokens.neutroQuente1,
      offWhite: tokens.offWhite,
      offWhite15: tokens.offWhite15,
      whiteSecondary: tokens.whiteSecondary,
      neutroFrio1: tokens.neutroFrio1,
      leve: tokens.leve,
      vermelhoAlerta: tokens.vermelhoAlerta,
      verdeSucesso: tokens.verdeSucesso,
      amareloAtencao: tokens.amareloAtencao,
      dropDownBorderColor: tokens.dropDownBorderColor,
      white: tokens.white,
      fullBlack: tokens.fullBlack,
      divider: tokens.divider,
      backgoundInput: tokens.backgoundInput,
      backgroundApp: tokens.backgroundApp,
      transparente: 'transparent',
      modalBackDropColor: tokens.modalBackDropColor,
      borderButton: tokens.borderButton,
      dividerCounter: tokens.dividerCounter,
      backgoundDivider: tokens.backgoundDivider,
      progressTextColor: tokens.progressTextColor,
      backgroundMenuOpened: tokens.backgroundMenuOpened,
      searchBarTextColor: tokens.searchBarTextColor,
      borderColorProduct: tokens.borderColorProduct
    },
    fonts: {
      nunitoBlack: 'Nunito-Black',
      nunitoBlackItalic: 'Nunito-BlackItalic',
      nunitoBold: 'Nunito-Bold',
      nunitoBoldItalic: 'Nunito-BoldItalic',
      nunitoExtraBold: 'Nunito-ExtraBold',
      nunitoExtraBoldItalic: 'Nunito-ExtraBoldItalic',
      nunitoExtraLight: 'Nunito-ExtraLight',
      nunitoExtraLightItalic: 'Nunito-ExtraLightItalic',
      nunitoItalic: 'Nunito-Italic',
      nunitoLight: 'Nunito-Light',
      nunitoLightItalic: 'Nunito-LightItalic',
      nunitoRegular: 'Nunito-Regular',
      nunitoSemiBold: 'Nunito-SemiBold',
      nunitoSemiBoldItalic: 'Nunito-SemiBoldItalic',
      reservaDisplayRegular: 'ReservaDisplay-Regular',
      reservaSerifBlack: 'ReservaSerif-Black',
      reservaSerifBlackItalic: 'ReservaSerif-BlackItalic',
      reservaSerifBold: 'ReservaSerif-Bold',
      reservaSerifBoldItalic: 'ReservaSerif-BoldItalic',
      reservaSerifItalic: 'ReservaSerif-Italic',
      reservaSerifLight: 'ReservaSerif-Light',
      reservaSerifLightItalic: 'ReservaSerif-LightItalic',
      reservaSerifMedium: 'ReservaSerif-Medium',
      reservaSerifMediumItalic: 'ReservaSerif-MediumItalic',
      reservaSerifRegular: 'ReservaSerif-Regular',
      reservaSerifThin: 'ReservaSerif-Thin',
      reservaSerifThinItalic: 'ReservaSerif-ThinItalic',
      reservaSansBlack: 'ReservaSans-Black',
      reservaSansBlackItalic: 'ReservaSans-BlackItalic',
      reservaSansBold: 'ReservaSans-Bold',
      reservaSansBoldItalic: 'ReservaSans-BoldItalic',
      reservaSansItalic: 'ReservaSans-Italic',
      reservaSansLight: 'ReservaSans-Light',
      reservaSansLightItalic: 'ReservaSans-LightItalic',
      reservaSansMedium: 'ReservaSans-Medium',
      reservaSansMediumItalic: 'ReservaSans-MediumItalic',
      reservaSansRegular: 'ReservaSans-Regular',
      reservaSansThin: 'ReservaSans-Thin',
      reservaSansThinItalic: 'ReservaSans-ThinItalic'
    },
    fontWeights: {
      light: 300,
      normal: 400,
      medium: 500,
      SemiBold: 600,
      bold: 700,
      black: 900
    },
    fontSizes: [4, 8, 12, 14, 16, 18, 20, 24, 27, 32, 48],
    radii: {
      pico: 2,
      nano: 5,
      xxxs: 10,
      xs: 15,
      sm: 20,
      pill: 500,
      infinity: 999999
    },
    borderWidths: {
      none: tokens.borderWidthNone,
      hairline: tokens.borderWidthHairline,
      thin: tokens.borderWidthThin,
      thick: tokens.borderWidthThick,
      heavy: tokens.borderWidthHeavy,
      borderDiscountTag: tokens.borderDiscountTag,
      stepper: 3
    }
  };
