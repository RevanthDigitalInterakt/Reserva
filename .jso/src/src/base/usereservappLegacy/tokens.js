  Object.defineProperty(exports, "__esModule", {
    value: true
  });
  exports.whiteSecondary = exports.white = exports.vermelhoRSV = exports.vermelhoFechadoRSV = exports.vermelhoAlerta = exports.verdeSucesso = exports.verde = exports.searchBarTextColor = exports.progressTextColor = exports.preto = exports.offWhite15 = exports.offWhite = exports.neutroQuente2 = exports.neutroQuente1 = exports.neutroFrio2 = exports.neutroFrio1 = exports.modalBackDropColor = exports.marinho = exports.leve = exports.fullBlack = exports.dropDownBorderColor = exports.dividerCounter = exports.divider = exports.borderWidthThin = exports.borderWidthThick = exports.borderWidthNone = exports.borderWidthHeavy = exports.borderWidthHairline = exports.borderDiscountTag = exports.borderColorProduct = exports.borderButton = exports.backgroundMenuOpened = exports.backgroundApp = exports.backgoundInput = exports.backgoundDivider = exports.amareloAtencao = undefined;
  var white = exports.white = '#FFF';
  var fullBlack = exports.fullBlack = '#000';

  // cores primarias
  var preto = exports.preto = '#333333';
  var vermelhoFechadoRSV = exports.vermelhoFechadoRSV = '#5A0000';
  var marinho = exports.marinho = '#13294B';
  var neutroFrio2 = exports.neutroFrio2 = '#8A8C8E';
  var neutroQuente2 = exports.neutroQuente2 = '#B1ACA8';
  var divider = exports.divider = '#D8D9DA';
  var dividerCounter = exports.dividerCounter = '#EAEAEA';
  var progressTextColor = exports.progressTextColor = '#979797';
  var searchBarTextColor = exports.searchBarTextColor = '#666666';

  // cores secundarias
  var vermelhoRSV = exports.vermelhoRSV = '#A00000';
  var verde = exports.verde = '#22372B';
  var neutroQuente1 = exports.neutroQuente1 = '#D7CFCA';
  var offWhite = exports.offWhite = '#D6D1C4';
  var offWhite15 = exports.offWhite15 = '#F9F8F6';
  var whiteSecondary = exports.whiteSecondary = '#FAFAFA';
  var neutroFrio1 = exports.neutroFrio1 = '#D8D9DA';
  var leve = exports.leve = '#F9E1DD';

  // cores do sistema
  var vermelhoAlerta = exports.vermelhoAlerta = '#EF1E1E';
  var verdeSucesso = exports.verdeSucesso = '#38A238';
  var amareloAtencao = exports.amareloAtencao = '#FEA528';
  var dropDownBorderColor = exports.dropDownBorderColor = '#DDDDDD';

  // larguras de borda
  var borderWidthNone = exports.borderWidthNone = 0;
  var borderWidthHairline = exports.borderWidthHairline = 1;
  var borderWidthThin = exports.borderWidthThin = 2;
  var borderWidthThick = exports.borderWidthThick = 4;
  var borderWidthHeavy = exports.borderWidthHeavy = 8;
  var borderDiscountTag = exports.borderDiscountTag = 12;

  // cores de borda
  var borderButton = exports.borderButton = '#333333';
  var borderColorProduct = exports.borderColorProduct = '#757575';

  // backgrounds
  var backgroundMenuOpened = exports.backgroundMenuOpened = '#F1F1F1';
  var backgroundApp = exports.backgroundApp = '#F9F9F9';
  var modalBackDropColor = exports.modalBackDropColor = '#AFAFAF';
  var backgoundInput = exports.backgoundInput = '#F0F0F0';
  var backgoundDivider = exports.backgoundDivider = '#F5F5F5';
