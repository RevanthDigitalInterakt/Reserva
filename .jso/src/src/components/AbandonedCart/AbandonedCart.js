  Object.defineProperty(exports, "__esModule", {
    value: true
  });
  exports.default = undefined;
  var _react = _$$_REQUIRE(_dependencyMap[0])(_$$_REQUIRE(_dependencyMap[1]));
  var _reactNative = _$$_REQUIRE(_dependencyMap[2]);
  var _ItemsCardWrapper = _$$_REQUIRE(_dependencyMap[0])(_$$_REQUIRE(_dependencyMap[3]));
  var _FooterAbandonedCart = _$$_REQUIRE(_dependencyMap[0])(_$$_REQUIRE(_dependencyMap[4]));
  var _HeaderAbandonedCart = _$$_REQUIRE(_dependencyMap[0])(_$$_REQUIRE(_dependencyMap[5]));
  var _testProps = _$$_REQUIRE(_dependencyMap[0])(_$$_REQUIRE(_dependencyMap[6]));
  function AbandonedCart() {
    return /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[7]).jsx)(_reactNative.SafeAreaView, {
      children: /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[7]).jsxs)(_reactNative.View, Object.assign({
        style: _$$_REQUIRE(_dependencyMap[8]).styles.container
      }, (0, _testProps.default)('abandoned_cart_container'), {
        children: [/*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[7]).jsx)(_HeaderAbandonedCart.default, {}), /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[7]).jsx)(_ItemsCardWrapper.default, {}), /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[7]).jsx)(_FooterAbandonedCart.default, {})]
      }))
    });
  }
  var _default = exports.default = AbandonedCart;
