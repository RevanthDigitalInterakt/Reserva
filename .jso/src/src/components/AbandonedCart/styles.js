  Object.defineProperty(exports, "__esModule", {
    value: true
  });
  exports.styles = undefined;
  var _reactNative = _$$_REQUIRE(_dependencyMap[0]);
  var styles = exports.styles = _reactNative.StyleSheet.create({
    container: {
      flex: 1,
      backgroundColor: _$$_REQUIRE(_dependencyMap[1]).COLORS.WHITE,
      marginTop: (0, _$$_REQUIRE(_dependencyMap[2]).scale)(25),
      margin: (0, _$$_REQUIRE(_dependencyMap[2]).scale)(12)
    }
  });
