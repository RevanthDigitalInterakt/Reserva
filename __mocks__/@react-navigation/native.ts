const navigate = jest.fn();

const useNavigation = () => ({
  navigate,
});

export {
  useNavigation,
};
