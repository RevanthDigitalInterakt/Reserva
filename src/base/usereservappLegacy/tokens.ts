export const white = '#FFF';
export const fullBlack = '#000';

// cores primarias
export const preto = '#333333';
export const vermelhoFechadoRSV = '#5A0000';
export const marinho = '#13294B';
export const neutroFrio2 = '#8A8C8E';
export const neutroQuente2 = '#B1ACA8';
export const divider = '#D8D9DA';
export const dividerCounter = '#EAEAEA';
export const progressTextColor = '#979797';
export const searchBarTextColor = '#666666';

// cores secundarias
export const vermelhoRSV = '#A00000';
export const verde = '#22372B';
export const neutroQuente1 = '#D7CFCA';
export const offWhite = '#D6D1C4';
export const offWhite15 = '#F9F8F6';
export const whiteSecondary = '#FAFAFA';
export const neutroFrio1 = '#D8D9DA';
export const leve = '#F9E1DD';

// cores do sistema
export const vermelhoAlerta = '#EF1E1E';
export const verdeSucesso = '#38A238';
export const amareloAtencao = '#FEA528';
export const dropDownBorderColor = '#DDDDDD';

// larguras de borda
export const borderWidthNone = 0;
export const borderWidthHairline = 1;
export const borderWidthThin = 2;
export const borderWidthThick = 4;
export const borderWidthHeavy = 8;
export const borderDiscountTag = 12;

// cores de borda
export const borderButton = '#333333';
export const borderColorProduct = '#757575';

// backgrounds
export const backgroundMenuOpened = '#F1F1F1';
export const backgroundApp = '#F9F9F9';
export const modalBackDropColor = '#AFAFAF';
export const backgoundInput = '#F0F0F0';
export const backgoundDivider = '#F5F5F5';
