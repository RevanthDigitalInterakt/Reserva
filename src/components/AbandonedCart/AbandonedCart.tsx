import React from 'react';
import { View, SafeAreaView } from 'react-native';
import { styles } from './styles';
import ItemsCardWrapper from './components/ItemsCardWrapper';
import FooterAbandonedCart from './components/FooterAbandonedCart';
import HeaderAbandonedCart from './components/HeaderAbandonedCart';
import testProps from '../../utils/testProps';

function AbandonedCart() {
  return (
    <SafeAreaView>
      <View style={styles.container} {...testProps('abandoned_cart_container')}>
        <HeaderAbandonedCart />
        <ItemsCardWrapper />
        <FooterAbandonedCart />
      </View>
    </SafeAreaView>
  );
}

export default AbandonedCart;
