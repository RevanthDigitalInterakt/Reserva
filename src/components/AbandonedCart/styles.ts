import { StyleSheet } from 'react-native';
import { COLORS } from '../../base/styles';
import { scale } from '../../utils/scale';

export const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: COLORS.WHITE,
    marginTop: scale(25),
    margin: scale(12),
  },
});
