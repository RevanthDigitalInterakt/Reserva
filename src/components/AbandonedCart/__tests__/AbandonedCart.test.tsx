import React from 'react';
import {
  render,
  screen,
} from '@testing-library/react-native';
import { MockedProvider } from '@apollo/client/testing';
import { ThemeProvider } from 'styled-components/native';
import { theme } from '../../../base/usereservappLegacy/theme';
import AbandonedCart from '../AbandonedCart';

const Component = (
  <ThemeProvider theme={theme}>
    <MockedProvider addTypename={false}>
      <AbandonedCart />
    </MockedProvider>
  </ThemeProvider>
);

describe('AbandonedCart', () => {
  beforeEach(async () => {
    jest.clearAllMocks();
  });

  it('snapshot', () => {
    render(Component);
    expect(screen.toJSON()).toMatchSnapshot();
  });

  it('should must be the rendering component', () => {
    render(Component);
    const mainContainer = screen.getAllByTestId('com.usereserva:id/abandoned_cart_container');
    const headerContainer = screen.getAllByTestId('com.usereserva:id/abandoned_cart_header_container');
    const carrousel = screen.getAllByTestId('com.usereserva:id/abandoned_cart_carrousel_content');
    const footerContainer = screen.getAllByTestId('com.usereserva:id/abandoned_cart_footer_container');

    expect(mainContainer).toBeTruthy();
    expect(headerContainer).toBeTruthy();
    expect(carrousel).toBeTruthy();
    expect(footerContainer).toBeTruthy();
  });
});
