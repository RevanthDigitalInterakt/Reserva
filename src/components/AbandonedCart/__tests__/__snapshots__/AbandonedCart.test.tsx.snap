// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AbandonedCart snapshot 1`] = `
<RCTSafeAreaView>
  <View
    accessibilityLabel="com.usereserva:id/abandoned_cart_container"
    style={
      {
        "backgroundColor": "#ffffff",
        "flex": 1,
        "margin": 28.125,
        "marginTop": 58.59375,
      }
    }
    testID="com.usereserva:id/abandoned_cart_container"
  >
    <View
      accessibilityLabel="com.usereserva:id/abandoned_cart_header_container"
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
      testID="com.usereserva:id/abandoned_cart_header_container"
    >
      <Text
        accessibilityLabel="com.usereserva:id/abandoned_cart_title"
        style={
          {
            "color": "#333333",
            "fontFamily": "Arial",
            "fontSize": 42.1875,
            "fontWeight": "400",
            "lineHeight": 48.515625,
          }
        }
        testID="com.usereserva:id/abandoned_cart_title"
      >
        Continue comprando
      </Text>
    </View>
    <View
      style={
        {
          "flex": 1,
        }
      }
    >
      <View
        collapsable={false}
        handlerTag={1}
        handlerType="PanGestureHandler"
        onGestureHandlerEvent={[Function]}
        onGestureHandlerStateChange={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        style={
          [
            {
              "overflow": "hidden",
            },
            {
              "height": 264,
              "width": 750,
            },
            {
              "alignItems": "center",
              "backgroundColor": "#ffffff",
            },
            {
              "flexDirection": "row",
            },
          ]
        }
        testID="com.usereserva:id/abandoned_cart_carrousel_content"
      />
    </View>
    <View
      style={
        [
          {
            "alignSelf": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "position": "relative",
          },
          {
            "width": 0,
          },
        ]
      }
    />
    <View
      accessibilityLabel="com.usereserva:id/abandoned_cart_button_finish_purchase"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
      testID="com.usereserva:id/abandoned_cart_button_finish_purchase"
    >
      <View
        accessibilityLabel="com.usereserva:id/abandoned_cart_footer_container"
        style={
          {
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "#303030",
            "borderColor": "#303030",
            "borderRadius": 4,
            "borderWidth": 1,
            "height": 40,
            "justifyContent": "center",
            "marginTop": 46.875,
            "width": 228,
          }
        }
        testID="com.usereserva:id/abandoned_cart_footer_container"
      >
        <Text
          style={
            {
              "color": "#ffffff",
              "fontFamily": "ReservaSans-Medium",
              "fontSize": 14,
              "fontWeight": "500",
              "lineHeight": 18.48,
              "textTransform": "uppercase",
            }
          }
          testID="com.usereserva:id/abandoned_cart_finish_purchase"
        >
          Finalizar compra
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
